rfcs/0005-fabric-elements-api.md

```diff
@@ -0,0 +1,43 @@
+# RFC: Fabric Elements API
+
+## Background
+
+When building our Web Component library (Fabric Elements) it's good to think of the overall picture of how the API should be structured. 
+With the right conventions, it would be easier to maintain consistency when adding new components, which should result in the library being easy to work with for both authors and users.
+
+This RFC attempts to start a discussion on how to structure the Fabric Elements API in a way that aims to minimise confusion and complexity of various components.
+
+## Concepts/Principles to discuss
+
+### Customization of components
+
+To what degree should web components be customized with respective attributes if they return just one main HTML element,
+e.g. when `<f-textarea>` returns `<textarea>`, `<f-button>` returns `<button>` or `<f-textfield>` returns `<input>`? 
```

Comment:
This is a bit on the side of your question, but I wonder if there's a minimal requirement for components we implement in Fabric. If a `f-button` web component only wraps a `button` element with a `class` attribute, does it add enough value to justify the added complexity?

The DOM is already capable of rendering `<button class="button button--primary">`, if that is all `<f-button type="primary">` would do. Adding a layer of JS around this will complicate SSR or force client-side rendering for behaviour that already exist.

Many web components will probably implements some extra behaviour, like form validation for input fields, and custom styling for checkboxes. And I clearly see the usefulness of having web components for that.

I'm sorry if this is off topic, but I would like to hear your thoughts on this. My main concern is that we use web components (and components in general) where it makes sense to encapsulate specific behaviour, and I believe there is a lower and upper bounds on the complexity it makes sense to implement with web components.

WDYT?

Reply:
👍